lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -323,10 +334,13 @@ sub engine_workit {
         return $error if $error;
     }
 
-    $vars{ASSETDIR}   //= OpenQA::Utils::assetdir();
-    $vars{CASEDIR}    //= OpenQA::Utils::testcasedir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
-    $vars{PRODUCTDIR} //= OpenQA::Utils::productdir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
+    my $casedir   = path(OpenQA::Utils::testcasedir($vars{DISTRI}, $vars{VERSION}, $shared_cache));
+    my $test_repo = $casedir->basename;
+    $vars{ASSETDIR} //= OpenQA::Utils::assetdir();
+    $vars{TESTREPO} = $test_repo;
```

Comment:
Please consider https://en.wikipedia.org/wiki/Test-driven_development as this can also help reviewers understand your intentions much better.

I assume you introduced the "TESTREPO" due to the problematic needle repo which can be outside CASEDIR in case of the "clone-custom-git-refspec" approach. I struggle with the name and meaning of "TESTREPO" though as for me it has just the same meaning as "CASEDIR", which is short for "test case directory".

Reply:
yes, 'TESTREPO' is used to find the needles. My intention about this parameter is `test repository`.